Replace api/version by api/status and add start time

alexeykazakov opened the pull request:
This PR replaces api/version by api/status and adds start time (UTC) to the response.

Format:
`{"build_time":"2016-09-28_03:00:00AM","commit":"<commit_SHA>","start_time":"2016-09-28_03:04:26AM"}`

Fixes #205

alexeykazakov commented:
This PR doesn't remove the healthcheck end point. Should we replace healthcheck by api/status?

sbose78 commented:
[test]

sbose78 commented:
@alexeykazakov Let's keep the /healthcheck and not remove it. It serves the purpose of checking database connectivity at the moment.

aslakknutsen commented:
@alexeykazakov You can remove healthcheck if you add the logic in it to the status endpoint :) (as mentioned in the issue description)

alexeykazakov commented:
OK. Thanks Shoubhik and Aslak. Just wanted to make sure we still want to combine healthcheck with API/status or keep them separate. I will update the pr later today.

alexeykazakov commented:
I added one more commit to the PR. Healtcheck is now replaced by/combined with api/status. @aslakknutsen @sbose78 please review.

aslakknutsen commented on status.go:
```diff
@@ -0,0 +1,35 @@
+
+	_, err := c.db.DB().Exec("select 1")
+	if err != nil {
+		return ctx.ServiceUnavailable(err);
```
Wondering if maybe we should return Res+Error in this case, not just the error. Thinking the build info is still useful even if it's failing.

alexeykazakov commented:
I have added one more commit. There is now an optional error message in the status media type
The status is returned always. If OK then w/o error message. If 503 then including the error.

For example when DB is unavailable (Status code: 503): `{"build_time":"0","commit":"0","error":"dial tcp 127.0.0.1:5432: getsockopt: connection refused","start_time":"2016-09-30_12:53:56AM"}`
For example when status is OK (Status code: 200): `{"build_time":"0","commit":"0","start_time":"2016-09-30_12:53:56AM"}`

And I rebased all commits.
@aslakknutsen take a look

sbose78 commented:
[test]

alexeykazakov commented:
The tests passes for me locally but I think there is a race condition and DB may be not available for status_test.go and this is why it failed on jenkins. I will try to refactor the test to make it stable.

alexeykazakov commented:
This is weird. I thought that maybe DB is nil in this test but DB is setup in workitem_whitebox_test.go in TestMain(). Both tests are in the same package so DB should be initialized by the time when TestShowStatusOK() is called. All works fine locally.
@baijum any idea what's going on? And is it possible to get more details (line numbers etc) of the failure?

baijum commented:
@alexeykazakov The tests are enabled to run in parallel. Since this is a DB integration test, you can remove that `t.Parallel()` statement.

alexeykazakov commented:
@baijum thanks! Run this test in parallel seems to be OK. But I should have mark this test to require a DB instead of UnitTest. It worked for me locally because my bash environment wasn't clean. Anyway I was able to reproduce this failure and fixed it in the new commit. Please re-run the check/tests. I believe it will ignore my [test] request bc I'm not an a member of the almighty team.

baijum commented:
[test]

baijum commented:
[test]

aslakknutsen commented:
Thank you @alexeykazakov !
